Introduce event producer following workqueue pattern

fridex opened the pull request.

fridex committed:
Introduce event producer following workqueue pattern

ghost commented:
Build failed.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/16/16/f3a99ff3cc3b1441ac631ebc6923d01b8b12eef3/check/thoth-coala/a4c6f8d/) : FAILURE in 1m 32s

fridex committed:
Do not use object from dict rest-client returns directly object

ghost commented:
Build succeeded.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/16/16/3360a0b57707532667b63deda85fa2e5ba4e5b92/check/thoth-coala/f549f43/) : SUCCESS in 1m 10s

goern reviewed (approved):
LGTM

fridex committed:
Instantiate OpenShift object in each process

If we will rely on code segment to be copied on fork(2), we can get TLS related
errors as the connection of TLS is kept instantiated in memory (copied on
fork(2)) and the HMAC check can fail. Make one instance per process to address
this issue.

ghost commented:
Build succeeded.

- [thoth-coala ](https://zuul.thoth-station.ninja/logs/16/16/e5d592ea14bf44307a6887ffbd1fb0f7e1aeaf57/check/thoth-coala/d714f0b/) : SUCCESS in 2m 00s

goern commented:
regate
